### ISSUE TITLE
Provenances are missing parameters

### ISSUE THREAD

**Issue by hyanwong** [tskit-dev/tsinfer#332]
In formats.py, all the provenances added by `record_provenance` in the modification methods (e.g. `subset()` `from-tree-sequence()`, etc) are added via:

```
xxx.record_provenance(command="my_cmd", **kwargs)
```

but often there are extra params (e.g. for subset, `individuals` and `sites`) that are not in `**kwargs`, and presumably should be added to the recorded provenance? If these are modified by the function (e.g. `if individuals is None: individuals = np.arange(sd.num_individuals)`), should we record the original value (`None`) or the one actually used in the final function (`np.arange()`), @jeromekelleher ?

**Comment by jeromekelleher** [comment]
Let's not worry about this now, we'll want to have a tskit standardised way of doing it and it's actually a lot of work to get right. But we've got bigger fish to try at the moment. Just recording that a particular command was used is a bonus for reproducibility.

**Comment by hyanwong** [comment]
Sure - leaving this issue open though, so we fix it correctly when we do.

**Comment by hyanwong** [comment]
I've hit this again just now. A user ran an inference and then wasn't sure what recombination rate they had used. It would have been super helpful to have been able to pull it out of the provenance data.

What's the status on automatically recording parameters in provenance data ATM? Is the idea to have a `tskit` function that will generate if somehow from within a function call? I'm guessing this is on @benjeffery 's to do list, but quite a long way down?

**Comment by jeromekelleher** [comment]
It's a long way down the priority list @hyanwong - we've already spent a lot more time on provenance than the value it has provided. We have lots of other things  to do, and this would be a bit rabbit hole.

I'd suggest putting in the parameters by hand for the functions that you're interested in.

**Comment by hyanwong** [comment]
> I'd suggest putting in the parameters by hand for the functions that you're interested in.

Right, that makes sense, thanks. I think the recombination rate and mismatch ratio are the important ones here, so that's easy enough. Did we decide how we wanted to embed large arrays (e.g. an msprime.RateMap) into the provenance data?

**Comment by jeromekelleher** [comment]
We decided not to do it, as it's much more likely to be a source of errors than to be helpful.

**Comment by hyanwong** [comment]
Great, thanks for clarifying. Perhaps if the object has a `str` rather than a `repr` method we should bung that in? That's assuming that the `str` method is meant to be a summary output of the object (which I think is what we do for tables, right?)

RateMap looks like it has a `str` function, so this would work immediately.

**Comment by jeromekelleher** [comment]
That's worse I think, as it violates the "you should be able to recover the output" rule. ``str`` is for human consumption.

As I said, we've already spent more time on this than we'll get back, we should just leave it.

**Comment by hyanwong** [comment]
> That's worse I think, as it violates the "you should be able to recover the output" rule. `str` is for human consumption.

True - I was thinking of it more as an additional "description" field attached to the parameter in the provenance, rather than the actual value, but you're probably right - too much hassle. I'll just check if the value's a scalar and if not, erm, well, I'm not sure actually.
